src/models.jl

```diff
@@ -19,19 +19,21 @@ end
 function dict_preds(::Val{:columnaccess}, func, target_table, idxsvec, weights)
     cols = Tables.columns(target_table)
     colnames = Tables.columnnames(cols)
-    dict_table = Dict(
+    dict = OrderedDict{Symbol, AbstractVector}(
         nm => func(weights, Tables.getcolumn(cols, nm), idxsvec) for nm in colnames
     )
+    dict_table = Tables.DictColumnTable(Tables.Schema(colnames, eltype.(values(dict))), dict)
     return dict_table
 end
 
 function dict_preds(::Val{:noncolumnaccess}, func, target_table, idxsvec, weights)
-     cols = Tables.dictcolumntable(target_table)
-     colnames = Tables.columnnames(cols)
-     dict_table = Dict(
-        nm => func(weights, Tables.getcolumn(cols, nm), idxsvec) for nm in colnames
-     )
-     return dict_table
+    cols = Tables.dictcolumntable(target_table)
+    colnames = Tables.columnnames(cols)
+    dict = OrderedDict{Symbol, AbstractVector}(
+       nm => func(weights, Tables.getcolumn(cols, nm), idxsvec) for nm in colnames
+    )
+    dict_table = Tables.DictColumnTable(Tables.Schema(colnames, eltype.(values(dict))), dict)
```

Comment:
Not sure `DictColumnTable` constructor is strictly public. Maybe we should be using [`Tables.dictcolumntable`](https://tables.juliadata.org/dev/#Tables.dictcolumntable) instead?

Reply:
The object `DictColumnTable` is public. So I don't see any harm in using the constructor.
The other alternative is to create my own table as `Tables.dictcolumntable` isn't a constructor, it a utility method that converts other tables into a `DictColumnTable`.

Reply:
I think it safest to construct your own table and use `dictcolumntable`.